### ISSUE TITLE
problem with install firmware, step 10

### ISSUE THREAD

**Issue by marioqwert9** [seemoo-lab/nexmon_csi#212]
I have problem in step 10. How to fix this error?

BUILDING DRIVER for kernel 5.4 brcmfmac_5.4.y-nexmon/brcmfmac.ko (details: log/driver.log)
make[1]: *** /lib/modules/5.4.81-v7l+/build: No such file or directory.  Stop.
make: *** [Makefile:108: brcmfmac.ko] Error 2

**Comment by zeroby0** [comment]
You don't have the right kernel headers.
You can get them as described here:
https://github.com/nexmonster/nexmon_csi/tree/pi-5.4.51-plus#get-kernel-headers

**Comment by marioqwert9** [comment]
I install this kernel patch but now i have this error. How to fix?

![187458226_168096235244813_2774460014669351452_n](https://user-images.githubusercontent.com/83959159/118890485-0c738280-b8ff-11eb-9b90-baba5ed2127d.jpg)

**Comment by marioqwert9** [comment]
I dont know is this the same error,how to fix ?
![187593139_1990844701067537_8605819163195489562_n](https://user-images.githubusercontent.com/83959159/118890636-53617800-b8ff-11eb-9f4f-3bcf172bf8d0.jpg)

**Comment by zeroby0** [comment]
Did you run `source setup_env.sh`? You have to do it everytime you are compiling Nexmon.

**Comment by marioqwert9** [comment]
Yes i run this in nexmon directory and than make. Than i go nexmon_csi folder and run make install-firmware... I have install Kernel 5.4.50.

**Comment by marioqwert9** [comment]
Can i record installation and post in on youtube. So you can tell me what i wrong do?

**Comment by zeroby0** [comment]
You don't have to. I have a build guide that I have tested a few hundred times and I know works. https://github.com/nexmonster/nexmon_csi/tree/pi-5.4.51-plus

Follow it **exactly** on a freshly formatted sd card, and your installation will work. Connect to the Pi with an ethernet cable if you're using ssh from computer though.

**Comment by marioqwert9** [comment]
Thank you, now i am on the last step with your tutorial. Tell me what you think with this Use ctrl-b d to detach, and tmux attach-session -t nexmon to re-attach.

**Comment by zeroby0** [comment]
Tmux opens a new virtual terminal inside your terminal. To get out of the terminal, use ctrl + b and then press d. To enter the tmux terminal again, use `tmux a`. It'd be a useful skill and would save time to google things :) https://www.google.com/search?q=tmux

**Comment by marioqwert9** [comment]
Ok, thank you very much. Is it need to to with new terminal, or not?
